packages/ruleset/src/functions/pagination-style.js

```diff
@@ -37,96 +45,35 @@ module.exports = function (pathObj, _opts, context) {
   logger.debug(`${ruleId}: checking pathItem at location: ${path.join('.')}`);
-
-  // The actual path string (e.g. '/v1/resources') will be the last element in 'path'.
-  const pathStr = path[path.length - 1];
-
-  // Retrieve this path item's 'get' operation.
-  const operation = pathItem.get;
-
-  // We'll bail out now if any of the following are true:
-  // 1. If the path string ends with a path param reference (e.g. '{resource_id}'
-  // 2. If the path item doesn't have a 'get' operation.
-  if (/}$/.test(pathStr) || !operation) {
-    logger.debug(`${ruleId}: 'get' operation is absent or excluded`);
-    return [];
-  }
-
-  // Next, find the first success response code.
-  const successCode = Object.keys(operation.responses || {}).find(code =>
-    code.startsWith('2')
-  );
-  if (!successCode) {
-    logger.debug(`${ruleId}: No success response code found!`);
-    return [];
-  }
-
-  // Next, find the json content of that response.
-  const content = operation.responses[successCode].content;
-  const jsonResponse = content && content['application/json'];
-
-  // If there's no response schema, then we can't check this operation so bail out now.
-  if (!jsonResponse || !jsonResponse.schema) {
-    logger.debug(`${ruleId}: No response schema found!`);
-    return [];
-  }
-
-  // Next, let's get the response schema (while potentially taking into account allOf).
-  const responseSchema = mergeAllOfSchemaProperties(jsonResponse.schema);
-  if (!responseSchema || !responseSchema.properties) {
-    logger.debug(`${ruleId}: Merged response schema has no properties!`);
-    return [];
-  }
-
-  // Next, make sure there is at least one array property in the response schema.
-  if (
-    !Object.values(responseSchema.properties).some(
-      prop => prop.type === 'array'
-    )
-  ) {
-    logger.debug(`${ruleId}: Response schema has no array property!`);
-    return [];
-  }
-
-  // Next, make sure this operation has parameters.
-  const params = operation.parameters;
-  if (!params) {
-    logger.debug(`${ruleId}: Operation has no parameters!`);
-    return [];
-  }
-
-  // Check to see if the operation defines a page token-type query param.
-  // This could have any of the names below.
-  const pageTokenParamNames = [
-    'start',
-    'token',
-    'cursor',
-    'page',
-    'page_token',
-  ];
-  const pageTokenParamIndex = params.findIndex(
-    param =>
-      param.in === 'query' && pageTokenParamNames.indexOf(param.name) !== -1
-  );
-
-  // Check to see if the operation defines an "offset" query param.
-  const offsetParamIndex = params.findIndex(
-    param => param.name === 'offset' && param.in === 'query'
-  );
-
-  // If the operation doesn't define a page token-type query param or an "offset" query param,
-  // then bail out now as pagination isn't supported by this operation.
-  if (pageTokenParamIndex < 0 && offsetParamIndex < 0) {
-    logger.debug(`${ruleId}: No start or offset query param!`);
```

Comment:
All of this code was moved so we could have one function to determine if a path has a paginated list operation or not